src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -55,18 +55,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String username = request.getParameter(USERNAME_PROPERTY_KEY);
     String bio = request.getParameter(BIO_PROPERTY_KEY);
     boolean edit = Boolean.parseBoolean(request.getParameter(EDIT_PROPERTY_KEY));
+    boolean isSameUsername = true;
+    UserProfile createdProfile = profileData.getUserProfile(email);
```

Comment:
I think this logic can be simplified a little bit and `isSameUsername` boolean can be removed.

You can use the new function that Bryan added to `ProfileDatabase` called `queryProfileByUsername()`.
If the return value is empty, then the username is not taken. If the return value has a UserProfile, you can compare that UserProfile's email with the current logged in email. If they are not the same, then the username is not taken. Something similar to below:

```
Optional<UserProfile> userProfile = profileData.queryProfileByUsername(username);
boolean isUserNameTaken = userProfile.isPresent() && !selfEmail.equals(userProfile.get().getEmail());
// TODO: if isUserNameTaken, return early with response.sendError()
```

Then you don't have to add individual checks inside the if/else edit branches below. It will help make the code cleaner and more readable.